### PR TITLE
Update documentation for Time Worked API

### PR DESCRIPTION
**Pull request by csaunders-humi**
https://gethumi.atlassian.net/browse/PRE-1689

- Adds documentation for additional income fields
- Adds documentation for inferred additional income entries

### CODE REVIEW

File: v1/payroll/employee_time_worked/post.md [hunk 1]
```diff
@@ -15,29 +15,75 @@ may be included for the employee.
...
+
+An `Inferred Additional Income` is provided for statutory holidays in which the hourly employee was provided a set wage. Inferred Additional Incomes
+will be applied to the employees primary rate that was active on the date of the statutory holiday.
+
+| Name       | Type     | Data Type              | Description                                                                                            |
+| ---------- | -------- | ---------------------- | ------------------------------------------------------------------------------------------------------ |
+| id         | required | uuid  (v4)             | UUID of the [CompanyAdditionalIncome](../../payroll/additional_incomes/get.md) (usually `holiday_pay`) |
+| start_time | required | datetime               | The start of the day for the statutory holiday (i.e. `2021-12-25 00:00:00 -05:00`                      |
+| end_time   | required | datetime               | The end of the day for the statutory holiday (i.e. `2021-12-25 23:59:59 -05:00`)                       |
+| amount     | required | money formatted string | The amount of money to be paid to the employee employee as compensation for the statutory holiday      |
```
Comment (claus-humi) [inline v1/payroll/employee_time_worked/post.md:73]: **suggestion:** ‏Right now accept hours if it's there, but calculate it from the amount if it's absent. Maybe we should add a section on calculating the hours from the amount (based on the rate)? It's somewhat not correct so the alternative would be to remove reading the hours all together from inferred_additional_income?

Comment (csaunders-humi) [reply v1/payroll/employee_time_worked/post.md:73]: @claus-humi do we accept hours right now? I thought we were always calculating it. If it's optional I can add that along with the caveat of how it works when it's absent.

**Comment by atuljha19** [comment]
very nicely captured!

File: v1/payroll/employee_time_worked/post.md (hunk 1 above)
Comment (claus-humi) [reply v1/payroll/employee_time_worked/post.md]: Looking at the code it seems like we do: https://github.com/Humi-HR/payroll/blob/master/app/services/time_worked_service/inferred_additional_income_creator_or_updater.rb#L46

Not sure if we should though?

File: v1/payroll/employee_time_worked/post.md [hunk 2]
```diff
@@ -15,29 +15,82 @@ may be included for the employee.
...
-| employee_id | required | uuid (v4)                      | The unique identifier for the employee                  |
-| time_worked | required | array of `Time Worked` entries | An array that contains at least one `Time Worked` entry |
+| Name                        | Type     | Data Type                                     | Description                                                                     |
+|-----------------------------|----------|-----------------------------------------------|---------------------------------------------------------------------------------|
+| employee_id                 | required | uuid (v4)                                     | The unique identifier for the employee                                          |
+| time_worked                 | required | array of `Time Worked` entries                | An array that contains at least one `Time Worked` entry                         |
+| reset                       | optional | boolean                                       | When set to true will destroy all previous time worked entries on open payrolls |
```
Comment (atuljha19) [inline v1/payroll/employee_time_worked/post.md:22]: we can mention that  currently we have set it to be true.
we did this to remove work load from Ameego

Comment (csaunders-humi) [reply v1/payroll/employee_time_worked/post.md:22]: @atuljha19 when you say "it is currently set to true" do you mean that if the value is absent we default to true?

Comment (atuljha19) [reply v1/payroll/employee_time_worked/post.md:22]: yes!

**Review by claus-humi — APPROVED** [review]

**Pull request merged**